### PR TITLE
Fix PV feedback on examples

### PR DESCRIPTION
**Pull request by rlnx**
PV reported some issues with examples, here is the fix + I've made makefiles for old and new examples more consistent.

### CODE REVIEW

File: examples/oneapi/cpp/makefile_lnx [hunk 1]
```diff
@@ -121,35 +122,28 @@ LOPTS := -L"$(DAAL_PATH)" \
          -L"$(TBB_PATH)" \
          $(LIBS)
 
-RES_DIR=_results/$(compiler)_$(_IA)_$(threading)_$(RES_EXT)
+RES_DIR=_results/$(compiler)_intel64_$(threading)_$(RES_EXT)
```
Comment (PetrovKP) [inline examples/oneapi/cpp/makefile_lnx:125]: From sln files https://github.com/oneapi-src/oneDAL/blob/master/examples/daal/cpp_sycl/daal_win.sln.tpl need to  remove support ia32?

File: examples/daal/cpp_sycl/makefile_lnx [hunk 2]
```diff
@@ -109,13 +109,16 @@ LOPTS := -Wl,--start-group -Wl,$(DAAL_LIB),$(EXT_LIB) $(DAAL_PATH)/libdaal_sycl.
 RES_DIR=_results/$(compiler)_$(_IA)_$(threading)_$(RES_EXT)
 RES = $(addprefix $(RES_DIR)/, $(if $(filter run, $(mode)), $(addsuffix .res ,$(example)), $(addsuffix .exe,$(example))))
 
+ifeq ($(compiler),clang)
+    gcc_toolchain := $(realpath $(dir $(shell which gcc))/..)
```
Comment (PetrovKP) [inline examples/daal/cpp_sycl/makefile_lnx:113]: By default, is this not so?

File: examples/oneapi/cpp/makefile_lnx [hunk 3]
```diff
@@ -19,7 +19,8 @@
 ##******************************************************************************
 
 help:
-	@echo "Usage: make {libia32|soia32|libintel64|sointel64|help}"
+	@echo "Usage: make {lib|so|libintel64|sointel64|help}"
+	@echo "lib is alias libintel64, so is alias for sointel64"
```
Comment (PetrovKP) [inline examples/oneapi/cpp/makefile_lnx:23]: Need to update
```make
##------------------------------------------------------------------------------
## examples of using:
##
## make libia32 example=assoc_rules_apriori_batch  - build  by Intel(R) C++ Compiler (as default)
##                                 and  run  association_rules  example  for 32-bit
##                                 applications, static linking
##
## make soia32 compiler=gnu      - build by GNU C++ compiler and run all examples
##                                 of Intel(R) DAAL for
##                                 32-bit applications, dynamic linking
##
## make libintel64 compiler=gnu  - build by GNU C++ compiler and run all example
##                                 of Intel(R) DAAL for

```

File: examples/daal/cpp_sycl/makefile_lnx (hunk 2 above)
Comment (rlnx) [reply examples/daal/cpp_sycl/makefile_lnx:113]: By default clang uses system GCC

**Pull request merged**